migrations/20210612_initial.sql

```diff
@@ -0,0 +1,104 @@
+CREATE TABLE profiles (
+  profile_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL,
+  properties TEXT NOT NULL -- JSON with user or contact profile
+);
+
+CREATE TABLE known_servers(
+  host TEXT NOT NULL,
+  port TEXT NOT NULL,
+  key_hash BLOB,
+  PRIMARY KEY (host, port)
+) WITHOUT ROWID;
+
+CREATE TABLE contacts (
+  contact_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL, -- display name set by local user
+  properties TEXT NOT NULL, -- JSON set by local user
+  profile_id INTEGER REFERENCES profiles, -- sent by remote contact
+  contact_status TEXT NOT NULL DEFAULT ''
+);
+
+CREATE TABLE contact_connections (
+  connection_id BLOB NOT NULL PRIMARY KEY,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  conn_level INTEGER NOT NULL DEFAULT 0,
+  via_conn BLOB REFERENCES contact_connections (connection_id),
+  conn_status TEXT NOT NULL DEFAULT ''
+) WITHOUT ROWID;
+
+CREATE TABLE contact_invitations (
+  invitation_id BLOB NOT NULL PRIMARY KEY,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  invitation_status TEXT NOT NULL DEFAULT ''
+) WITHOUT ROWID;
+
+CREATE TABLE groups (
+  group_id INTEGER NOT NULL PRIMARY KEY, -- local group ID
+  display_name TEXT NOT NULL, -- shared
+  properties TEXT NOT NULL -- shared JSON group profile
+);
+
+CREATE TABLE group_members ( -- group members, nullable fields are NULL for the local user
+  group_id INTEGER NOT NULL REFERENCES groups ON DELETE RESTRICT,
+  member_id BLOB NOT NULL, -- shared member ID, unique per group
+  member_role TEXT NOT NULL DEFAULT '', -- owner, admin, moderator, ''
+  member_status TEXT NOT NULL DEFAULT '', -- inv | con | full | off
+  contact_id INTEGER REFERENCES contacts ON DELETE RESTRICT,
+  invited_by INTEGER REFERENCES contacts ON DELETE RESTRICT,
+  connection_id BLOB UNIQUE,
```

Comment:
why duplicate? isn't it already available by `contact_id` from `contacts`?

Reply:
no, it's another connection specifically for group messages.

We thought to combine at some point, but it's impossible to identify group then - it requires either global group ID or pairwise IDs for all member pairs - which is what connections provide already.

So it's a separate connection.